src/components/accordion/daikin-accordion.ts

```diff
@@ -0,0 +1,199 @@
+
+    "after:block",
+    "after:w-4",
+    "after:h-4",
+    "after:m-auto",
+    "after:top-0",
+    "after:right-4",
+    "after:bottom-0",
+    "after:absolute",
+    "after:i-daikin-accordion-chevron",
+    "after:transition-all",
+    "after:rotate-0",
+  ],
+  {
+    variants: {
+      disabled: {
+        enabled: ["after:text-[#828282]"],
+        disabled: ["text-[#DCDCDC]", "after:text-[#DCDCDC]"],
+      },
+    },
+    defaultVariants: {
+      disabled: "enabled",
+    },
+  }
+);
+
+const cvaContent = cva(["pt-2", "pb-[24px]", "px-[20px]", "text-sm"]);
+
+const animationOptionBase = {
+  duration: 1000,
+  easing: "ease-in-out",
+};
+
+const contentDefault = {
+  height: 0,
+};
+const contentVisible = (content: HTMLElement) => [
+  {
+    height: `${content.offsetHeight}px`,
+  },
+];
+
+/**
+ * Primary UI component for user interaction
+ */
+@customElement("daikin-accordion")
+export class DaikinAccordion extends LitElement {
+  static override readonly styles = css`
+    ${unsafeCSS(tailwindStyles)}
+
+    :host {
+      --accordion-summary-chevron-rotate: 0;
+      display: block;
+      width: 100%;
+      max-width: 400px;
+    }
+  `;
+
+  /**
+   * Heading of accordion
+   */
+  @property({ type: String })
+  override title = "";
+
+  /**
+   * Whether the accordion is open
+   */
+  @property({ type: Boolean, reflect: true })
+  open = false;
+
+  /**
+   * Whether the accordion is disabled
+   */
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  @state()
+  dataOpen = false;
+
+  private _visibleContents(e: PointerEvent) {
+    const content = (e.target as HTMLElement).nextElementSibling as HTMLElement;
+    e.preventDefault();
+
+    if (this.disabled) {
+      return;
+    }
+
+    if (this.dataOpen) {
+      this.dataOpen = false;
+      const animation = content.animate(
+        [...contentVisible(content), contentDefault],
+        animationOptionBase
+      );
+
+      animation.onfinish = () => {
+        this.open = this.dataOpen;
+      };
+    } else {
+      this.dataOpen = true;
+      this.open = this.dataOpen;
+      content.animate(
+        [contentDefault, ...contentVisible(content)],
+        animationOptionBase
+      );
+    }
+  }
+
+  override render() {
+    const accordionDetailsClassName = cvaDetails({
+      disabled: this.disabled ? "disabled" : "enabled",
+    });
+
+    const accordionSummaryClassName = cvaSummary({
+      disabled: this.disabled ? "disabled" : "enabled",
+    });
+
+    const accordionContentClassName = cvaContent();
+
+    return html`<details
+      class=${accordionDetailsClassName}
+      ?open=${this.open}
+      ?data-open=${this.dataOpen}
+      aria-disabled=${this.disabled}
+    >
+      <summary
+        class=${accordionSummaryClassName}
+        @click=${this._visibleContents}
+        tabindex=${this.disabled ? -1 : 0}
+      >
+        ${this.title}
+      </summary>
+      <div>
+        <div class=${accordionContentClassName}>
```

Comment:
One of the advantages of using TailwindCSS is separation of concerns.
The closer the element and style are described, the easier it is to see what the element will look like.

```suggestion
        <div class="pt-2 pb-[24px] px-[20px] text-sm">
```